client/rest/src/plugins/metadata/metadataRoutes.js

```diff
@@ -19,10 +19,14 @@
  * along with Catapult.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+const { MetalSeal } = require('./metal');
 const catapult = require('../../catapult-sdk/index');
 const merkleUtils = require('../../routes/merkleUtils');
 const routeResultTypes = require('../../routes/routeResultTypes');
 const routeUtils = require('../../routes/routeUtils');
+const NodeCache = require('node-cache');
```

Comment:
Sorry, I don't know much about it, so if you direct me to it, I'll set it up that way.

Regarding the cache, do I limit the maximum number of keys to be stored?

When saving the cache, do I limit the size of the file?

I thought that limiting the total data size is not a good idea, as it would require calculating the total cached files each time.

If there is a limit on the number or size of keys, does it need to be set in the config file as in the comment above?

@Wayonb @Jaguar0625

Reply:
In the first place, the purpose is to CACHE accesses to the same key instantaneously.

It might be better to set the key limit to 1 (actually 2 is better since textSection is also cached) and also set the size limit.

I don't know what the appropriate size limit is, but I think the size for a mobile camera is around 4 MB.
Sorry, I have no idea if this is appropriate.

Reply:
> I thought that limiting the total data size is not a good idea, as it would require calculating the total cached files each time.

This depends on how much memory is used. The memory used will be related to the number of MetalId created across the chain. Are you saying there will not be more than one MetalId?

I don't know the use cases for metal, so I don't know how much memory will be used. Usually, limiting memory prevents a node from running out of memory.

The reason I asked cause I noticed ``node-cache`` disable the max keys by default.
 
https://www.npmjs.com/package/node-cache#options

> maxKeys: (default: -1) specifies a maximum amount of keys that can be stored in the cache. If a new item is set and the cache is full, an error is thrown and the key will not be saved in the cache. -1 disables the key limit.

Reply:
> Are you saying there will not be more than one MetalId?

No, Metal IDs may be created as many times as users create them as Metal.
There is no size limit because we only concatenate metadata.

How about acquiring the size of cached data each time in this way, and if it exceeds the cache limit specified in rest.json as well as the TTL, how about not caching?

```js
const totalSize = cache.keys().reduce((total, key) => total + cache.get(key).length, 0);
```
sorry, I didn't know that `node-cache` has maxKeys as optional.
if I set a size limit as above, key may not need to be restricted.

Reply:
I added limit size. if total cached size + data size < limitSize data is cached.
how do you think?

Reply:
> sorry, I didn't know that node-cache has maxKeys as optional.

No problems.  We are all learning 😄 

> How about acquiring the size of cached data each time in this way, and if it exceeds the cache limit specified in rest.json as well as the TTL, how about not caching?

I think this is fine not sure the overhead of calculating the ``totalSize`` if the metalId is not in the cache.
@Jaguar0625 your thoughts?

Reply:
Thanks.  Looks good 👍🏾